web-marketplace/src/pages/BuyCredits/BuyCredits.Form.tsx

```diff
@@ -165,46 +182,120 @@ export const BuyCreditsForm = ({
       stripe?: Stripe | null,
       elements?: StripeElements | null,
     ) => {
-      const { retirementReason, country, stateProvince, postalCode } = values;
-      const {
-        sellOrders: selectedSellOrders,
-        email,
-        name,
-        savePaymentMethod,
-        createAccount: createActiveAccount,
-        // subscribeNewsletter, TODO
-        // followProject,
-      } = data;
+      const sellOrders = await refetchSellOrders();
+      const requestedSellOrders = findMatchingSellOrders(
+        data,
+        sellOrders?.map(normalizeToUISellOrderInfo),
+      );
+      const currentAvailableCredits = requestedSellOrders.reduce(
+        (credits, sellOrder) => credits + Number(sellOrder.quantity),
+        0,
+      );
+      const sellCanProceed =
+        data.creditsAmount && data.creditsAmount < currentAvailableCredits;
+      const partialCreditsAvailable =
+        data.creditsAmount && data.creditsAmount > currentAvailableCredits;
 
-      if (selectedSellOrders)
-        purchase({
-          paymentOption,
-          selectedSellOrders,
-          retiring,
-          retirementReason,
-          country,
-          stateProvince,
-          postalCode,
+      if (sellCanProceed) {
+        const { retirementReason, country, stateProvince, postalCode } = values;
+        const {
+          sellOrders: selectedSellOrders,
           email,
           name,
           savePaymentMethod,
-          createActiveAccount,
-          paymentMethodId,
-          stripe,
-          elements,
-          confirmationTokenId,
+          createAccount: createActiveAccount,
+          // subscribeNewsletter, TODO
+          // followProject,
+        } = data;
+
+        if (selectedSellOrders)
+          purchase({
+            paymentOption,
+            selectedSellOrders,
+            retiring,
+            retirementReason,
+            country,
+            stateProvince,
+            postalCode,
+            email,
+            name,
+            savePaymentMethod,
+            createActiveAccount,
+            paymentMethodId,
+            stripe,
+            elements,
+            confirmationTokenId,
+          });
+      } else if (partialCreditsAvailable) {
+        setUserCanPurchaseCredits({
+          openModal: true,
+          amountAvailable: currentAvailableCredits,
+        });
+      } else {
+        setUserCanPurchaseCredits({
+          openModal: true,
+          amountAvailable: 0,
         });
```

Comment:
reading the dev note on https://www.figma.com/design/BTuUv6QXY4GbliZcXCe8RJ/Fiat-payments?node-id=365-62773&node-type=instance&m=dev, I believe we should also check for other sell orders for the project, right now it's only checking the amount still available for the already selected sell orders